File: Projet_flutter_backend/src/models/client.model.js
const { DataTypes } = require("sequelize");
const { sequelize } = require("../db");
const Client = sequelize.define(
  "Client",
  {
    code: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    raisonSociale: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    adresse: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    codePostal: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    region: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    creeLe: {
      type: DataTypes.DATEONLY,
      allowNull: true,
    },
    regime: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    matriculeFiscal: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    identifiantUnique: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    contact: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    derniereFacturation: {
  type: DataTypes.DATEONLY,
  allowNull: true,
},
  },
  {
    tableName: "clients",
    timestamps: true,
  }
);

module.exports = Client;

File: Projet_flutter/lib/application/users/model/client_model.dart
class ClientModel {
  final int? id;
  final String? code;
  final String? raisonSociale;
  final String? adresse;
  final String? codePostal;
  final String? region;
  final String? creeLe;
  final String? regime;
  final String? matriculeFiscal;
  final String? identifiantUnique;
  final String? contact;
  final String? derniereFacturation;

  ClientModel({
    this.id,
    this.code,
    this.raisonSociale,
    this.adresse,
    this.codePostal,
    this.region,
    this.creeLe,
    this.regime,
    this.matriculeFiscal,
    this.identifiantUnique,
    this.contact,
    this.derniereFacturation,
  });

  factory ClientModel.fromJson(Map<String, dynamic> json) {
    return ClientModel(
      id: json['id'] is int ? json['id'] : int.tryParse('${json['id']}'),
      code: json['code']?.toString(),
      raisonSociale: json['raisonSociale']?.toString(),
      adresse: json['adresse']?.toString(),
      codePostal: json['codePostal']?.toString(),
      region: json['region']?.toString(),
      creeLe: json['creeLe']?.toString(),
      regime: json['regime']?.toString(),
      matriculeFiscal: json['matriculeFiscal']?.toString(),
      identifiantUnique: json['identifiantUnique']?.toString(),
      contact: json['contact']?.toString(),
      derniereFacturation: json['derniereFacturation']?.toString(),
    );
  }
}